src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -0,0 +1,311 @@
+//! Implements the segmented output port area.
+use crate::prelude::*;
+
+use enso_frp as frp;
+use enso_frp;
+use ensogl::data::color;
+use ensogl::display::Attribute;
+use ensogl::display::Buffer;
+use ensogl::display::Sprite;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::*;
+use ensogl::display::traits::*;
+use ensogl::display;
+use ensogl::gui::component::Animation;
+use ensogl::gui::component;
+
+use crate::node::NODE_SHAPE_PADDING;
+use crate::node::NODE_SHAPE_RADIUS;
+
+
+
+// =================
+// === Constants ===
+// =================
+// TODO: These values should be in some IDE configuration.
+
+const BASE_SIZE           : f32 = 0.5;
+const HIGHLIGHT_SIZE      : f32 = 1.0;
+const SEGMENT_GAP_WIDTH   : f32 = 2.0;
+
+
+
+// ==============
+// === Shapes ===
+// ==============
+
+/// The port area shape is based on a single shape that gets offset to show a different slice for
+/// each segment. Each shapes represents a window of the underlying shape.
+pub mod port_area {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (style:Style, grow:f32, shape_width:f32, offset_x:f32, padding:f32, opacity:f32) {
+            let width  : Var<Distance<Pixels>> = shape_width.into();
+            let height : Var<Distance<Pixels>> = "input_size.y".into();
```

Comment:
Why width is not computed the same way height is ?

Reply:
There are two widths here: the whole shape and the visible slice.